AK/BigInts.h

```diff
@@ -0,0 +1,487 @@
+/*
+ * Copyright (c) 2021, Leon Albrecht <leon2002.la@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Checked.h>
+#include <AK/Concepts.h>
+#include <AK/Format.h>
+#include <AK/NumericLimits.h>
+#include <AK/StdLibExtraDetails.h>
+#include <AK/StdLibExtras.h>
+#include <AK/String.h>
+#include <AK/StringBuilder.h>
+
+namespace AK {
+
+template<typename T>
+class UBigInt;
+template<typename T>
+constexpr bool Detail::IsIntegral<UBigInt<T>> = true;
+template<typename T>
+constexpr bool Detail::IsUnsigned<UBigInt<T>> = true;
+template<typename T>
+constexpr bool Detail::IsSigned<UBigInt<T>> = false;
+
+template<typename T>
+class UBigInt {
+    // this sadly cannot be `require(...)`-d
+    static_assert(sizeof(T) >= sizeof(u64), "Creating BigInts of types smaller than u64 is useless");
+    static_assert(IsUnsigned<T>, "Creating BigInts of signed or non-integer types is not supported");
```

Comment:
same as the one above?

Reply:
this wont work for similar reasons as above, and because the IsUnsinged<T> isn't overloaded at that point in time to accept the recursion